backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/logging/impl/ElasticsearchLog.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * Copyright Red Hat Inc. and Hibernate Authors
+ */
+
+package org.hibernate.search.backend.elasticsearch.logging.impl;
+
+import static org.jboss.logging.Logger.Level.TRACE;
+
+import org.hibernate.search.util.common.logging.impl.MessageConstants;
+
+import org.jboss.logging.annotations.LogMessage;
+import org.jboss.logging.annotations.Message;
+import org.jboss.logging.annotations.MessageLogger;
+import org.jboss.logging.annotations.ValidIdRange;
+import org.jboss.logging.annotations.ValidIdRanges;
+
+@MessageLogger(projectCode = MessageConstants.PROJECT_CODE)
+@ValidIdRanges({
+		@ValidIdRange(min = MessageConstants.BACKEND_ES_ID_RANGE_MIN, max = MessageConstants.BACKEND_ES_ID_RANGE_MAX),
+})
+public interface ElasticsearchLog
+		extends AnalyzerLog, CommonFailureLog, ConfigurationLog, ElasticsearchClientLog, ElasticsearchRequestLog,
+		ElasticsearchSpecificLog, IndexingLog,
+		MappingLog, QueryLog, SchemaExportLog, VersionLog {
```

Comment:
I'll probably better reply to some of these before finishing the renaming to `INSTANCE` 😄 

> If you really want to go that far, wouldn't it make sense to have all of these as nested interfaces under ElasticsearchLog, at least, so that it's easier to maintain -- especially when it comes to the ID range?

Yeah that was my initial thought as well to keep them all as inner interfaces but:

>Cyclic inheritance involving 'org. hibernate. search. engine. logging. impl. EngineLog'

If we want the main interface to extend these subs... and I think we do want it to extend them as that way, we rely on the AP to verify that there's no message ID overlap... 

Note that the category collector will only include the categories where the interface has the `@LogMessage(level = ` so if we have interfaces with just exceptions, those won't be listed as logging categories.